core/src/main/scala/kafka/server/LogDirFailureChannel.scala

```diff
@@ -38,7 +41,8 @@ class LogDirFailureChannel(logDirNum: Int) {
    * If the given logDir is not already offline, add it to the
    * set of offline log dirs and enqueue it to the logDirFailureEvent queue
    */
-  def maybeAddLogFailureEvent(logDir: String): Unit = {
+  def maybeAddLogFailureEvent(logDir: String, msg: => String, t: IOException): Unit = {
```

Comment:
Since the exception is optional, it should be be `Option[IOException]`. We should name it `exception` instead of `t`. Alternatively, we could have an overloaded method that doesn't take an exception.